File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:

    string longestCommonPrefix(vector<string>& strs) {
        string sub = strs[0]; int size = strs.size(); 
        for(int i =1; i< strs.size();i++){
            while(strs[i].find(sub) != 0){
                sub = sub.substr(0,sub.size()-1);
            }
            if(sub.size() == 0) return "";
        }
        return sub;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length == 1) return strs[0];
        String s = strs[0];

        for(int i = s.length()-1;i>=0;i--){
           String ss = s.substring(0,i+1);
           int cnt = 0;
           for(int j=0;j<strs.length;j++){
            if(i>=strs[j].length()) break;
             if( strs[j].substring(0, i + 1).equals(ss)){
                cnt++;
             }
           }
           if(cnt == strs.length) return ss;
        }
        return "";
    }
}